install_opi5.sh

```diff
@@ -4,55 +4,89 @@ if id "$1" >/dev/null 2>&1; then
     echo 'user found'
 else
     echo "creating pi user"
-    useradd pi -b /home
+    useradd pi -m -b /home -s /bin/bash
     usermod -a -G sudo pi
-    mkdir /home/pi
-    chown -R pi /home/pi
 fi
 echo "pi:raspberry" | chpasswd
 
-apt-get update
+apt-get update --quiet
+
+before=$(df --output=used / | tail -n1)
+# clean up stuff
+echo 'Purging snaps'
+# get rid of snaps
+rm -rf /var/lib/snapd/seed/snaps/*
+rm -f /var/lib/snapd/seed/seed.yaml
+apt-get purge --yes --quiet lxd-installer lxd-agent-loader
+apt-get purge --yes --quiet snapd
+
+# remove bluetooth daemon
+apt-get purge --yes --quiet bluez
+
+apt-get --yes --quiet autoremove
+
+after=$(df --output=used / | tail -n1)
+freed=$(( before - after ))
+
+echo "Freed up $freed KiB"
+
+# run Photonvision install script
 wget https://git.io/JJrEP -O install.sh
 chmod +x install.sh
 
-sed -i 's/# AllowedCPUs=4-7/AllowedCPUs=4-7/g' install.sh
+sed -i 's/# AllowedCPUs=4-7/AllowedCPUs=0-7/g' install.sh
 
-./install.sh -n -q
+./install.sh -m -q
 rm install.sh
 
+echo "Installing additional things"
 
-# Remove extra packages 
-echo "Purging extra things"
-# apt-get remove -y gdb gcc g++ linux-headers* libgcc*-dev
-# apt-get remove -y snapd
-apt-get autoremove -y
+apt-get install --yes --quiet network-manager net-tools libatomic1
 
+# let netplan create the config during cloud-init
+rm -f /etc/netplan/00-default-nm-renderer.yaml
 
-echo "Installing additional things"
-sudo apt-get update
-apt-get install -y network-manager net-tools libatomic1
+# set NetworkManager as the renderer in cloud-init
+cp -f ./OPi5_CIDATA/network-config /boot/network-config
 
-apt-get install -y libc6 libstdc++6
+# add customized user-data file for cloud-init
+cp -f ./OPi5_CIDATA/user-data /boot/user-data
 
-# cat > /etc/netplan/00-default-nm-renderer.yaml <<EOF
-# network:
-#   renderer: NetworkManager
+# tell NetworkManager not to wait for the carrier on ethernet, which can delay boot
+# when the coprocessor isn't connected to the ethernet
+# cat > /etc/NetworkManager/conf.d/50-ignore-carrier.conf <<EOF
+# [main]
+# ignore-carrier=*
```

Comment:
Why's this still commented out?

Reply:
Ignoring the carrier may change current behavior on the robot. I'm not sure that it's a problem, but I'd like to leave it commented out until I have a chance to test on a robot network to make sure that nothing unexpected happens.